sw/device/tests/sim_dv/gpio_test.c

```diff
@@ -0,0 +1,295 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_gpio.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/log.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_plic.h"
+#include "sw/device/lib/handler.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/pinmux.h"
+#include "sw/device/lib/runtime/check.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/testing/test_main.h"
+#include "sw/device/lib/testing/test_status.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_gpio_t gpio;
+static dif_plic_t plic;
+
+static const uint32_t kNumGpios = 32;
+static const uint32_t kGpiosMask = 0xffffffff;
+// TODO #2484: GPIO pins 16-19 are special inputs - do not touch them.
+static const uint32_t kGpiosAllowedMask = 0xfff0ffff;
+
+static const uint32_t kNumChipGpios = 16;
+static const uint32_t kChipGpiosMask = 0xffff & kGpiosAllowedMask;
+
+// These indicate the GPIO pin irq expected to fire, declared volatile since
+// they are used by the interrupt handler.
+static volatile uint32_t exp_gpio_pin_irq;
```

Comment:
`exp` -> `expected`.

Reply:
There is no list of approved abbreviations, and indeed very little agreement on them in existing DIF reviews.

I have no problem with `exp_` and `obs_` as conventions for expected/observed in test code.